internal/extender/demand.go

```diff
@@ -135,8 +133,7 @@ func newDemand(pod *v1.Pod, units []demandapi.DemandUnit) (*demandapi.Demand, er
 			Name:      demandName,
 			Namespace: pod.Namespace,
 			Labels: map[string]string{
-				SparkAppIDLabel:    appID,
-				instanceGroupLabel: instanceGroup,
+				SparkAppIDLabel: appID,
```

Comment:
@ashrayjain I am assuming that this instanceGroup label that was set by scheduler on the demand object was not actually needed, and instead the instanceGroup on the DemandSpec below is what matters - does that match your understanding?

Reply:
yes, that is correct. we use the value from the spec and don't expect the label to be present.